dakota/src/lib.rs

```diff
@@ -307,19 +307,22 @@ impl Dakota {
     fn initialize_platform() -> Result<(Box<dyn Platform>, th::Thundr, (f32, f32))> {
         #[cfg(feature = "sdl")]
         if std::env::var("DISPLAY").is_ok() || std::env::var("WAYLAND_DISPLAY").is_ok() {
-            if let Ok(sdl) = platform::SDL2Plat::new() {
-                let mut sdl: Box<dyn Platform> = Box::new(sdl);
-                match Self::init_thundr(&mut sdl) {
-                    Ok(thundr) => match thundr.get_dpi() {
-                        Ok(dpi) => {
-                            log::error!("Using SDL2 Platform Backend");
-                            return Ok((sdl, thundr, dpi));
-                        }
+            match platform::SDL2Plat::new() {
+                Ok(sdl) => {
+                    let mut sdl: Box<dyn Platform> = Box::new(sdl);
+                    match Self::init_thundr(&mut sdl) {
+                        Ok(thundr) => match thundr.get_dpi() {
+                            Ok(dpi) => {
+                                log::error!("Using SDL2 Platform Backend");
+                                return Ok((sdl, thundr, dpi));
+                            }
+                            Err(e) => log::error!("Failed to create SDL2 backend: {:?}", e),
```

Comment:
This predates your change but I would change this message to "Failed to get DPI from SDL" to reflect what the error actually takes place during.